WaiterRobot/Ui/Order/OrderScreen.swift

```diff
@@ -75,4 +55,69 @@ struct OrderScreen: View {
         }
         .handleSideEffects(of: viewModel, navigator)
     }
+
+    @ViewBuilder
+    private func currentOder(
+        _ currentOrderArray: KotlinArray<OrderItem>
+    ) -> some View {
+        let currentOrder = Array(currentOrderArray)
```

Comment:
I am not sure if there is another possibility, but I assume this could be a performance "issue". As with each recomposition, a new Array will be created by appending all the items from the Kotlin array to the new array.
If the array only contains the references to the Kotlin Object, then this will probably not be noticeable. But if there is some copy in the background, this would be a quite heavy operation.
Maybe we can create an Iterable from it which reuses the Kotlin iterator?

Reply:
We can not work around this as this is only regenerated when the value changes and we need to copy it. Furthermore, Arrays are value types in Swift 😅